### ISSUE TITLE
sbrk_base is the same as sbrk_end

### ISSUE THREAD

**Issue by gsingh93** [cloudburst/libheap#7]
Not sure if this is a bug or me just not understanding something, but I keep seeing output like:
```
sbrk_base 0x601000
chunk     0x601000         0x410        (inuse)
chunk     0x601410         0x20bf0      (top)
sbrk_end  0x601000
```

when running `heap -l`. Note that `sbrk_base` is the same as `sbrk_end`. I'm using Arch Linux, but a friend tested it out on Ubuntu and had the same issue. Can you replicate this?

**Comment by cloudburst** [comment]
It's because the libheap struct malloc_state is outdated and doesn't handle PER_THREAD arenas.  Will need to be patched.

http://sourceware.org/git/?p=glibc.git;a=commitdiff;h=425ce2edb9d11cc1ff650fac16dfbc450241896a
```c
+#ifdef PER_THREAD
+  /* Linked list for free arenas.  */
+  struct malloc_state *next_free;
+#endif
```

**Comment by cloudburst** [comment]
The value seems to have changed from system_mem to max_system_mem.  Fixed in 5c6b68dda30e0f89801a5b4c3219529aacc24532
